packages/core/src/PartialExecutionUtils/__tests__/findStartNodes.test.ts

```diff
@@ -0,0 +1,376 @@
+// NOTE: Diagrams in this file have been created with https://asciiflow.com/#/
+// If you update the tests, please update the diagrams as well.
+// If you add a test, please create a new diagram.
+//
+// Map
+// 0  means the output has no run data
+// 1  means the output has run data
+// ►► denotes the node that the user wants to execute to
+// XX denotes that the node is disabled
+// PD denotes that the node has pinned data
+
+import { type IPinData, type IRunData } from 'n8n-workflow';
+import { createNodeData, toITaskData } from './helpers';
+import { findStartNodes, isDirty } from '../findStartNodes';
+import { DirectedGraph } from '../DirectedGraph';
+
+describe('isDirty', () => {
+	test("if the node has pinned data it's not dirty", () => {
+		const node = createNodeData({ name: 'Basic Node' });
+
+		const pinData: IPinData = {
```

Comment:
Something for later on: Would be nice to have a factory function to create `IPinData`

Reply:
Created an issue for it: https://linear.app/n8n/issue/PAY-1944/create-factory-function-to-create-ipindata